File: A1/A1-025/A1-025.cpp
#include<iostream>
#include<map>
using namespace std;

map<char, string> NUMS = {{'A', "ace"}, {'J', "jack"}, {'Q', "queen"}, {'K', "king"}};
map<char, string> GROUPS = {{'D', "diamonds"}, {'H', "hearts"}, {'S', "spades"}, {'C', "clubs"}};

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input
    string card;
    cin >> card;
    // Make name all uppercase
    for(int i = 0; i < card.size(); i++)
    {
        card[i] = toupper(card[i]);
    }
    
    // Output card number
    if(card.size() == 2)
    {
        // Card number are [A, J, Q, K]
        char num = card[0];
        if(NUMS.find(num) != NUMS.end()) { cout << NUMS[num]; }
        // Card number are 2-9
        else { cout << num; }
    }
    // Card number are 10
    else if(card.size() == 3) { cout << 10; }
    
    // Output card group
    char group = card[card.size() - 1];
    cout << " of " << GROUPS[group];
    return 0;
}

File: A1/A1-025/A1-025.py
# --------------------------------------------------
# File Name : A1-025.py
# Problem   : Cards
# Author    : Worralop Srichainont
# Date      : 2026-02-08
# --------------------------------------------------

# Dict of card value and card groups
CARD_VALUES = {
    "A": "ace",
    "2": "2",
    "3": "3",
    "4": "4",
    "5": "5",
    "6": "6",
    "7": "7",
    "8": "8",
    "9": "9",
    "10": "10",
    "J": "jack",
    "Q": "queen",
    "K": "king",
}
CARD_GROUPS = {"D": "diamonds", "H": "hearts", "S": "spades", "C": "clubs"}

# Input card and find its value and group
card = input().upper()
card_value = card[:-1]
card_group = card[-1]

# Output card name
print(f"{CARD_VALUES[card_value]} of {CARD_GROUPS[card_group]}")
